### PR TITLE
Cross validation

### PR DESCRIPTION
**Pull request by jaybrandon**

### CODE REVIEW

File: src/train_cv.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            shuffle=False,
+            num_workers=4,
+            pin_memory=torch.cuda.is_available(),
+        )
+
+        model = init_model(dataset).to(device)
+
+        criterion = nn.CrossEntropyLoss()
+
+        # run.watch(model, criterion, log="all", log_freq=100)  # Makes training slower :(
```
Comment (codewithlaurin) [inline src/train_cv.py]: Remove comment

File: src/train_cv.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
+import json
+import os
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from sklearn.metrics import f1_score
+from sklearn.model_selection import StratifiedKFold
+from sympy.logic.boolalg import true
```
Comment (codewithlaurin) [inline src/train_cv.py]: dis used?

File: src/train_cv.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+from torch.utils.data import Subset
+from torch.utils.data.dataloader import DataLoader
+from torchvision import models
+from torchvision.datasets.folder import ImageFolder
+from tqdm import tqdm
+
+import wandb
+from dataset import DATA_TRANSFORMS, PROJECT_ROOT, get_train_dataset
+
+K_FOLDS = 5
```
Comment (codewithlaurin) [inline src/train_cv.py:19]: config file 🥺👉👈

File: src/train_cv.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+    fold_params = []
+    fold_scores = []
+
+    for fold, (train_idx, val_idx) in tqdm(
+        enumerate(kfold.split(dataset.samples, dataset.targets)),
+        "KFold",
+        config["k_folds"],
+        leave=False,
+    ):
+        train_folder = get_train_dataset(DATA_TRANSFORMS["train"])
```
Comment (codewithlaurin) [inline src/train_cv.py]: loading data set on each fold

File: src/dataset.py [hunk 5]
```diff
@@ -31,14 +55,29 @@
...
+    "Cherry_(including_sour)___healthy",
 ]
 
 
 def get_data_path(img_type=IMAGE_TYPE_COLOR):
     return kagglehub.dataset_download("abdallahalidev/plantvillage-dataset") + img_type
 
 
-def split_data(test_size=0.2, random_state=27, out_dir=SPLIT_OUT_DIR):
+def split_data(test_size=0.2, random_state=42, out_dir=SPLIT_OUT_DIR):
```
Comment (codewithlaurin) [inline src/dataset.py:80]: Irgendwie chammer dä shit au globbally setze 

something like this:
torch.manual_seed(42)
np.random.seed(42)

**Review by codewithlaurin — CHANGES_REQUESTED** [review]
Well done, thank you for your effort <3

File: src/train_cv.py (hunk 3 above)
Comment (jaybrandon) [reply src/train_cv.py:19]: constants will be removed for sweep

**Review by codewithlaurin — APPROVED** [review]
